### ISSUE TITLE
Add visualization of predicted bounding boxes on Weights & Biases

### ISSUE THREAD

**Issue by soumik12345** [Deci-AI/super-gradients#1136]
### 🚀 Feature Request

Add feature to **optionally** perform inference with the model while training on a small subset of the validation data at a fixed interval and log them on Weights & Biases. This helps us gain further insight into how the model performs on specific examples from the validation set.

This feature is part of similar Weights & Biases integrations like YOLOv5, YOLOv8, Keras, etc.

- [This report](https://wandb.ai/av-team/bdd100k-perception/reports/BoundingBoxDebugger-23-06-01-11-16-31---Vmlldzo0NTI4MTE2) shows a media panel with the bounding boxes logged at the end of every training epoch for [YOLOv5](https://github.com/ultralytics/yolov5). This slider lets us visualize the predicted bounding boxes logged after every epoch.
- [This report](https://wandb.ai/parambharat/YOLOv8/reports/Sample-Detections--Vmlldzo0NTY5MDk3) shows the same as an interactive [`wandb.Table`](https://docs.wandb.ai/guides/data-vis) for [YOLOv8](https://github.com/ultralytics/ultralytics).
- [This report](https://wandb.ai/ayush-thakur/mmdetection/reports/How-To-Use-Weights-Biases-With-MMDetection--VmlldzoyMTM0MDE2#visualize-dataset-and-model-prediction) shows the same for [mmdetection](https://github.com/open-mmlab/mmdetection) as an interactive [`wandb.Table`](https://docs.wandb.ai/guides/data-vis).

### Proposed Solution

Possible ways to implement:

- Implement as a [`Callback`](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/utils/callbacks/base_callbacks.py#L94) by overriding the [`on_validation_batch_end`](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/utils/callbacks/base_callbacks.py#L289).
- Implement as part of the [`WandBSGLogger`](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/common/sg_loggers/wandb_sg_logger.py#L30).

**Comment by soumik12345** [comment]
@ofrimasad @BloodAxe 
I would love to discuss this!
